pkg/vgmanager/cleanup.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vgmanager
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/red-hat-storage/lvm-operator/controllers"
+	topolvmv1 "github.com/topolvm/topolvm/api/v1"
+	lvmCMD "github.com/topolvm/topolvm/pkg/lvmd/cmd"
+	crc "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// vgCleanup removes all volume groups existing in lvmd config file and on every vg
+// removal it also removed config file
+func vgCleanup(r *VGReconciler, ctx context.Context) error {
+
+	// get limited number of logical volume CRs to verify their presence
+	lvs := &topolvmv1.LogicalVolumeList{}
+	if err := r.Client.List(ctx, lvs, &crc.ListOptions{Limit: int64(1)}); err == nil {
+		// do not proceed with cleanup if there are logicalVolumes
```

Comment:
Aren't we suppose to check if the LV CR is mapped with the VG somehow?  Right now we are aborting VG deletion if at least one LV CR is found.   This  won't be very useful in case of the cleaning up individual VGs (for example user removes a VG from the spec)

Reply:
- that can be tweaked but I don't want to do that in this PR
- the flow will be more or less same however in controller itself there'll be some extra checks as well

Reply:
so the question would be even if the user is deleting the entire cluster, do we want to skip all VG deletion even if one LV CR is available? or we delete whatever VGs can be deleted in the cluster?

Reply:
Yes, until we have implementation of device selector and specific vg deletion, I'd like to abort any deletion.

When we implement the option 2, we can spend time on extra processing.

Reply:
It would be better to have a vgCleanupAll function which calls a vgCleanup (vgname) for each vg.

Reply:
the current function does the same, it'll remove volume group one by one based on vgName in lvmd config file